### PR TITLE
OCPBUGS-23216: Update to v1.26.7

### PR DESCRIPTION
**Pull request by bertinatto**
Created with:

```
git checkout -B rebase-v1.26.7 v1.26.7
git merge -s ours ocp/release-4.13
git cherry-pick 4cbe6cdd62db6d4a210009d516a0bad395e41b35
git cherry-pick 590e6a8fd1fcd4009e12be8aff7a3f90edb731fa
(manually fixing conflicts)
```

**Diff with upstream:** 

https://github.com/kubernetes-sigs/azuredisk-csi-driver/compare/v1.26.7...bertinatto:rebase-v1.26.7

**Notable changes:**

- [release-1.26] fix: ensure Kubernets conformant format for location  in [#1990](https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1990)
- [release-1.26] feat: add copy snapshot in cross region directly in [#1844](https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1844)
- fix: detach disk failure when there is throttling ([#1758](https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1758))
- fix: PremiumV2_LRS caching mode issue ([#1794](https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1794))
- fix: panic when allow-empty-cloud-config is set ([#1699](https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1699))

Note: the main goal of this rebase is to pull in https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1755.

### CODE REVIEW

File: .ci-operator.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-8-release-golang-1.19-openshift-4.12
+  tag: rhel-8-release-golang-1.19-openshift-4.13
```
Comment (jsafrane) [inline .ci-operator.yaml]: This looks wrong. We do not use 4.13 imagesin 4.12. Did you target the right branch?

**Comment by duanwei33** [comment]
Seem no much custom-facing feature introduced and CI looks good.
/label qe-approved
/label cherry-pick-approved

**Pull request merged**